StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/PollingViewModel.swift

```diff
@@ -32,6 +34,18 @@ class PollingViewModel {
             return Date().addingTimeInterval(60 * 5) // 5 minutes
         case .blik:
             return Date().addingTimeInterval(60) // 60 seconds
+        case .paynow:
+            return Date().addingTimeInterval(60 * 60) // 1 hour
+        default:
+            fatalError("Polling deadline has not been implemented for \(paymentMethodType)")
+        }
+    }()
+    lazy var maxRetries: Int = {
+        switch paymentMethodType {
+        case .UPI, .blik:
+            return 12
+        case .paynow:
+            return 1000 // PayNow has a very long expiration, allow a high number of retries
```

Comment:
What happens after maxRetries is hit?

Reply:
It looks like we do exponential backoff here, is that right? If so, why? Shouldn't we be polling at regular intervals (probably 1 second, like Checkout)?  Exponential backoff would mean we'd soon be waiting incredibly long before updating the PI.

Reply:
Exponential backoff was the original design decision for polling in mobile, which started with UPI. https://docs.google.com/document/d/1HbbLaf0sJ869BZmBoxsfxYjG31ymBk-Zdrl9g_5s62U/edit

I'm open to simplifying this by
1. Retrying on a short interval like 1 second for all polling LPMs
2. Removing retry count in favor of just using the count down timer. IIRC retry count is needed for the exponential back off formula.

Reply:
Making the customer wait exponential amounts of time before they can finish checking out for the sake of battery/network conservation isn't the right tradeoff IMO, especially given that we're only polling when the customer is in the app, staring at the spinner. Retrying every second seems fine, and if we wanted to do some backoff, it should slowly go to a max of like every 10 seconds.


Also the retry code looks broken today, although I think we should rewrite it. 
```
pow(Double(1 + maxRetries - retryCount), Double(2))
```

At maxRetries = 1000, retryCount = 0, this gives a delay of 1001^2 = a million seconds and grows smaller as we retry.

Reply:
@yuki-stripe I updated the polling logic to:
1. Poll every second
2. Remove the max retry count

I re-ran all the tests for PayNow and the experience is a bit better now, much snappier. Same for BLIK and UPI.

Reply:
Would this also mean the retry logic for 3DS2 is broken? It's the same formula: https://github.com/stripe/stripe-ios/blob/master/StripePayments/StripePayments/Source/PaymentHandler/STPPaymentHandler.swift

Reply:
No, but it's not obvious at all.  `retryCount` is the wrong name in that file.  It starts equal to `maxRetries` and goes _down_.  It should really be called `retriesRemaining`.